### ISSUE TITLE
Checkout using git `~` syntax

### ISSUE THREAD

**Issue by paraseba** [earth-mover/icechunk#273]
For example, checkout `main~~`. Of course, the checkout will be translated to a snapshot, so it doesn't allow commit to the same branch.

**Comment by rabernat** [comment]
I love the feature, but I hate the syntax! I'd prefer

```python
repo.checkout(branch="main", relative=-1)
```

or something more literate.

**Comment by paraseba** [comment]
Yeah, any syntax would work ... the proposed one is what git does, but we don't need to do the same.
